### PR TITLE
New util error funcs

### PR DESCRIPTION
**Pull request by grgustaf**
These are from my board-api branch but not specific to it.

### CODE REVIEW

**Comment by brianjjones** [comment]
LGTM +1

File: src/zjs_util.c [hunk 1]
```diff
@@ -77,14 +77,30 @@ void zjs_obj_add_object(jerry_value_t parent, jerry_value_t child,
...
+                                 const char *name)
+{
+    // requires: obj is an existing JS object
+    //  effects: creates a new readonly field in parent named name, set to str
+    jerry_value_t jname = jerry_create_string((const jerry_char_t *)name);
+    jerry_property_descriptor_t pd;
+    jerry_init_property_descriptor_fields(&pd);
+    pd.is_writable = false;
+    pd.is_value_defined = true;
+    pd.value = jerry_create_string((const jerry_char_t *)str);
```
Comment (jimmy-huang) [inline src/zjs_util.c]: you don't need to call jerry_release_value() on pd.value? or the jerry_free_property_descriptor_fields() does it?

Comment (grgustaf) [inline src/zjs_util.c]: Yeah, from API description: "Free fields of property descriptor (setter, getter and value)."

**Comment by jimmy-huang** [comment]
+1

**Pull request merged**